lib/model-builder.js

```diff
@@ -332,6 +339,38 @@ ModelBuilder.prototype.define = function defineClass(className, properties, sett
     dataSource.attach(this);
   };
 
+  // initialize getMergePolicy if not done already via model heritance
+  ModelClass.getMergePolicy = ModelClass.getMergePolicy || function() {
+    const mergePolicy = {
+      description: {replace: true}, // string or array
+      options: {patch: true}, // object
+      properties: {patch: true}, // object
+      hidden: {replace: false}, // array
+      protected: {replace: false}, // array
+      indexes: {patch: true}, // object
+      methods: {patch: true}, // object
+      mixins: {patch: true}, // object
+      // validations: {patch: true}, // object // NOTE: not implemented
+      relations: {patch: true},
+      scope: {replace: true}, // object
+      scopes: {patch: true}, // object
+      acls: {rank: true}, // array
+      // this setting controls which child model property's value allows deleting
+      // a base model's property
+      __delete: null,
+      // this setting controls the default merge behaviour for settings not defined
+      // in the mergePolicy specification
+      __default: {replace: true},
+    };
+
+    // make sure we take settings of current model with `this`
+    // model might have no settings in case it is an instance of builtin ModelBaseClass
+    return (this.settings || {}).mergePolicy || mergePolicy;
+  };
+
+  // ModelClass.base might not have getMergePolicy() if it is an instance of builtin ModelBaseClass
+  ModelClass.base.getMergePolicy = ModelClass.base.getMergePolicy || ModelClass.getMergePolicy;
```

Comment:
> how do we make sure `base` always have `.getMergePolicy` defined?

I think it should be enough to define `ModelBaseClass.getMergePolicy()` in `lib/model.js`, since all models should be always inheriting from `ModelBaseClass` (or a child model that has already transitively inherited from `ModelBaseClass`).

Alternatively, if there is no easy way for achieving that, then I'd rather fix the code that's currently requiring an existing `.getMergePolicy` method to support the case when such method is not defined.

Regarding the example with `A`, `B` and `C`: I think the problem is that in the current implementation, if `B` and `C` have different merge policies, then the merge policy used by `base.getMergePolicy` will depend on the order in which `B` and `C` are defined. Assuming that models are defined in lexicographical order, when you rename model `B` to model `D`, suddenly the default merge policy will change from what model `B`/`D` provides to what model `C` provides.

What if we changed this line to `assert` that `base.getMergePolicy` is already defined? If all existing tests are passing, then we should be good. If there are some tests that start failing, then that should give us more insights into possible inheritance hierarchies.

```diff
- ModelClass.base.getMergePolicy = ModelClass.base.getMergePolicy || ModelClass.getMergePolicy;
+ assert(ModelClass.base.getMergePolicy, `The base class ${ModelClass.modelName}` does not provide getMergePolicy(). Most likely it is not inheriting from juggler's DefaultModelBaseClass, which is an incorrect usage of the framework.`);
```